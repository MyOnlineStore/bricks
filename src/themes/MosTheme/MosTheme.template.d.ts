import { FunctionComponent } from 'react';
declare const MosTheme: FunctionComponent;
export default MosTheme;
