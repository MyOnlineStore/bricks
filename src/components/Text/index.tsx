import styled from '../../utility/_styled';
import SeverityType from '../../types/_SeverityType';

type TextVariantStyleType = {
    fontFamily: string;
    fontSize: string;
    fontWeight: string;
    lineHeight: {
        default: string;
        compact: string;
    };
};

type TextThemeType = {
    default: {
        color: string;
    };
    variant: {
        small: TextVariantStyleType;
        regular: TextVariantStyleType;
        large: TextVariantStyleType;
        extraLarge: TextVariantStyleType;
        display: TextVariantStyleType;
    };
    strong: {
        fontWeight: string;
    };
    severity: {
        error: string;
        success: string;
        info: string;
        warning: string;
    };
};

type PropsType = {
    variant?: 'small' | 'regular' | 'large' | 'extraLarge' | 'display';
    severity?: SeverityType;
    textAlign?: 'left' | 'right' | 'center' | 'justify';
    compact?: boolean;
    strong?: boolean;
};

const Text = styled.p<PropsType>`
    color: ${({ severity, theme }): string => (severity ? theme.Text.severity[severity] : theme.Text.default.color)};
    font-family: ${({ variant, theme }): string =>
        !variant ? theme.Text.variant.regular.fontFamily : theme.Text.variant[variant].fontFamily};
    font-size: ${({ variant, theme }): string =>
        !variant ? theme.Text.variant.regular.fontSize : theme.Text.variant[variant].fontSize};
    font-weight: ${({ variant, strong, theme }): string => {
        if (strong) {
            return theme.Text.strong.fontWeight;
        } else if (variant && !strong) {
            return theme.Text.variant[variant].fontWeight;
        }

        return theme.Text.variant.regular.fontWeight;
    }};
    line-height: ${({ variant, compact, theme }): string => {
        if (compact && variant) {
            return theme.Text.variant[variant].lineHeight.compact;
        } else if (!compact && variant) {
            return theme.Text.variant[variant].lineHeight.default;
        } else if (compact && !variant) {
            return theme.Text.variant.regular.lineHeight.compact;
        }

        return theme.Text.variant.regular.lineHeight.default;
    }};
    text-align: ${({ textAlign }): string => (textAlign ? textAlign : '')};
    margin: 0;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
`;

export default Text;
export { TextThemeType, TextVariantStyleType, PropsType };
