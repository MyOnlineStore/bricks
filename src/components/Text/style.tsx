import _R from 'react';
import { StyledComponentClass as _S } from 'styled-components';
import _T from '../../types/ThemeType';
import styled, { withProps } from '../../utility/styled';
import { PropsType } from '.';

type TextVariantStyleType = {
    fontFamily: string;
    fontSize: string;
    fontWeight: string;
    lineHeight: {
        default: string;
        compact: string;
    };
};

type TextThemeType = {
    variant: {
        small: TextVariantStyleType;
        regular: TextVariantStyleType;
        large: TextVariantStyleType;
        XL: TextVariantStyleType;
        display: TextVariantStyleType;
    };
    strong: {
        fontWeight: string;
    };
    severity: {
        default: string;
        error: string;
        success: string;
        info: string;
        warning: string;
    };
};

const StyledParagraph = withProps<PropsType, HTMLParagraphElement>(styled.p)`
    color: ${({ severity, theme }): string =>
        !severity ? theme.Text.severity.default : theme.Text.severity[severity]};
    font-family: ${({ variant, theme }): string =>
        !variant ? theme.Text.variant.regular.fontFamily : theme.Text.variant[variant].fontFamily};
    font-size: ${({ variant, theme }): string =>
        !variant ? theme.Text.variant.regular.fontSize : theme.Text.variant[variant].fontSize};
    font-weight: ${({ variant, strong, theme }): string => {
        if (strong) {
            return theme.Text.strong.fontWeight;
        } else if (variant && !strong) {
            return theme.Text.variant[variant].fontWeight;
        } else {
            return theme.Text.variant.regular.fontWeight;
        }
    }};
    line-height: ${({ variant, compact, theme }): string => {
        if (compact && variant) {
            return theme.Text.variant[variant].lineHeight.compact;
        } else if (!compact && variant) {
            return theme.Text.variant[variant].lineHeight.default;
        } else if (compact && !variant) {
            return theme.Text.variant.regular.lineHeight.compact;
        } else {
            return theme.Text.variant.regular.lineHeight.default;
        }
    }};
    text-align: ${({ textAlign }): string => (textAlign ? textAlign : '')};
    margin: 0;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;

    // This makes sure text wraps in IE:
    flex: 1;
    `;

const StyledSpan = StyledParagraph.withComponent('span');

export { StyledSpan, StyledParagraph, TextThemeType, TextVariantStyleType };
