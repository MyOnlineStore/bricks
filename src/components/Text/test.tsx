import React from 'react';
import Text from '.';
import { shallowWithTheme } from '../../utility/styled/testing';
import 'jest-styled-components';

describe('Text', () => {
    it('should render text with default styling', () => {
        const component = shallowWithTheme(<Text>Default text</Text>);

        expect(component.dive()).toMatchSnapshot();
    });

    it('should render text with strong styling', () => {
        const component = shallowWithTheme(<Text strong>Strong text</Text>);

        expect(component.dive()).toHaveStyleRule('font-weight', '700');
    });

    it('should render text with descriptive styling', () => {
        const component = shallowWithTheme(<Text descriptive>Descriptive text</Text>);

        expect(component.dive()).toHaveStyleRule('color', '#88979d');
    });

    it('should render text with compact styling', () => {
        const component = shallowWithTheme(<Text compact>Descriptive text</Text>);

        expect(component.dive()).toHaveStyleRule('line-height', '1.25');
    });

    it('should render a span when inline is set to true', () => {
        const component = shallowWithTheme(<Text inline>Descriptive text</Text>);

        expect(component.dive().find('span').length).toBe(1);
    });
});
