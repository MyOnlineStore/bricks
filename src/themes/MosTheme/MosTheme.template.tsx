import React, { StatelessComponent } from 'react';
import { ThemeProvider } from 'styled-components';
import theme from './MosTheme.theme';

type PropsType = {

};

const MosTheme:StatelessComponent<PropsType> = (props):JSX.Element => {
    return (
        <ThemeProvider theme={theme}>
            {props.children}
        </ThemeProvider>
    );
};

export default MosTheme;
export { PropsType };
