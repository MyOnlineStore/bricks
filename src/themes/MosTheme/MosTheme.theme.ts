import ThemeType from '../../types/ThemeType';

const bodyFont = 'Source Sans Pro, sans-serif';
const headingFont = 'Melbourne, sans-serif';

const fontSize = {
    larger: '36px',
    large: '28px',
    medium: '22px',
    base: '14px',
};

const green = {
    light: '#7ce88a',
    base: '#6bde78',
    dark: '#5bd16a',
    darker: '#4cc55b',
};

const silver = {
    light: '#fff',
    base: '#f8f9fb',
    medium: '#edeff2',
    dark: '#e4e7ec',
    darker: '#dbdfe6',
    extradark: '#d2d7e0',
};

const grey = {
    lighter: '#a6aab3',
    light: '#88979d',
    medium: '#55656d',
    base: '#333740',
    dark: '#21252b',
};

const red = {
    light: '#ff245e',
    base: '#ed2157',
    medium: '#dc1b4f',
    dark: '#c81746',
};

const yellow = {
    light: '#ffda5b',
    base: '#ffce28',
    medium: '#fcc200',
    dark: '#edb107',
};

const theme:ThemeType = {
    Button: {
        common: {
            borderRadius: '0',
            borderWidth: '1px',
            fontSize: fontSize.base,
            fontWeight: '400',
            fontFamily: bodyFont,
        },
        primary: {
            idle: {
                backgroundColor: green.dark,
                borderColor: green.dark,
                color: silver.light,
            },
            hover: {
                backgroundColor: green.dark,
                borderColor: green.dark,
                color: silver.light,
            },
            focus: {
                backgroundColor: green.dark,
                borderColor: green.dark,
                color: silver.light,
            },
            active: {
                backgroundColor: green.dark,
                borderColor: green.dark,
                color: silver.light,
            },
        },
        secondary: {
            idle: {
                backgroundColor: silver.medium,
                borderColor: silver.extradark,
                color: grey.base,
            },
            hover: {
                backgroundColor: silver.medium,
                borderColor: silver.extradark,
                color: grey.base,
            },
            focus: {
                backgroundColor: silver.medium,
                borderColor: silver.extradark,
                color: grey.base,
            },
            active: {
                backgroundColor: silver.medium,
                borderColor: silver.extradark,
                color: grey.base,
            },
        },
        warning: {
            idle: {
                backgroundColor: yellow.medium,
                borderColor: yellow.medium,
                color: silver.light,
            },
            hover: {
                backgroundColor: yellow.medium,
                borderColor: yellow.medium,
                color: silver.light,
            },
            focus: {
                backgroundColor: yellow.medium,
                borderColor: yellow.medium,
                color: silver.light,
            },
            active: {
                backgroundColor: yellow.medium,
                borderColor: yellow.medium,
                color: silver.light,
            },
        },
        destructive: {
            idle: {
                backgroundColor: red.base,
                borderColor: red.base,
                color: silver.light,
            },
            hover: {
                backgroundColor: red.base,
                borderColor: red.base,
                color: silver.light,
            },
            focus: {
                backgroundColor: red.base,
                borderColor: red.base,
                color: silver.light,
            },
            active: {
                backgroundColor: red.base,
                borderColor: red.base,
                color: silver.light,
            },
        },
        disabled:{
            color: grey.base,
            backgroundColor: silver.medium,
            stripingColor: silver.extradark,
        },
    },
    FlatButton: {
        idle: {
            borderColor: 'transparent',
            borderRadius: '0',
            borderWidth: '1px',
            color: grey.dark,
            fontFamily: bodyFont,
            fontSize: fontSize.base,
        },
        hover: {
            borderColor: silver.extradark,
            color: grey.dark,
        },
        focus: {
            borderColor: silver.extradark,
            color: grey.dark,
        },
        active: {
            borderColor: silver.extradark,
            color: grey.dark,
        },
    },
    Notification: {
        common: {
            fontFamily: bodyFont,
            fontSize: fontSize.medium,
        },
        error: {
            color: red.dark,
            backgroundColor: '#ffbdce',
            borderColor: '#ff6c93',
        },
        warning: {
            color: '#bb8d0a',
            backgroundColor: '#fff0be',
            borderColor: yellow.base,
        },
        success: {
            color: '#31953d',
            backgroundColor: '#d2f5d6',
            borderColor: green.base,
        },
        info: {
            color: '#4e82bb',
            backgroundColor: '#dde7f2',
            borderColor: '#c8d2dd',
        },
    },
    Text:{
        color: grey.base,
        fontFamily: bodyFont,
        fontSize: fontSize.base,
    },
    Heading:{
        color: grey.base,
        fontFamily: headingFont,
        fontSize: fontSize.large,
        fontWeight: '400',
    },
    PriceTag: {
        default: {
            color: grey.base,
            fontFamily: bodyFont,
            fontSize: fontSize.large,
            fontWeight: '400',
            super: {
                fontSize: '.7em',
                fontWeight: '400',
            },
        },
        action: {
            fontSize: fontSize.medium,
            fontWeight: '400',
            color: grey.light,
            fontFamily: bodyFont,

            super: {
                fontSize: '.7em',
                fontWeight: '400',
            }
        }
    },
    SubHeading:{
        color: grey.light,
        fontFamily: bodyFont,
        fontSize: fontSize.medium,
        fontWeight: '400',
    }
};

export default theme;
