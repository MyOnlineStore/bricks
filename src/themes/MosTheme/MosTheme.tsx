import React, { StatelessComponent } from 'react';
import { ThemeProvider } from 'styled-components';
import themeSettings from './settings';

type PropsType = {

};

const MosTheme:StatelessComponent<PropsType> = (props):JSX.Element => {
    return (
        <ThemeProvider theme={themeSettings}>
            {props.children}
        </ThemeProvider>
    );
};

export default MosTheme;
export { PropsType };
