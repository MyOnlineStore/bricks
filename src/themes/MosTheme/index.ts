export { default as default } from './MosTheme';
