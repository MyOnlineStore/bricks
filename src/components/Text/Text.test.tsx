import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import Text from '.';
import MosTheme from '../../themes/MosTheme';

describe('Text',  () => {
    it('should render text with default styling', () => {
        const component = shallow(
            <MosTheme>
                <Text>Default text</Text>
            </MosTheme>
        ).dive();

        expect(toJson(component)).toMatchSnapshot();
    });

    it('should render text with strong styling', () => {
        const component = shallow(
            <MosTheme>
                <Text strong={true}>Strong text</Text>
            </MosTheme>
        ).dive();

        expect(toJson(component)).toMatchSnapshot();
    });

    it('should render text with descriptive styling', () => {
        const component = shallow(
            <MosTheme>
                <Text descriptive={true}>Descriptive text</Text>
            </MosTheme>
        ).dive();

        expect(toJson(component)).toMatchSnapshot();
    });
});
