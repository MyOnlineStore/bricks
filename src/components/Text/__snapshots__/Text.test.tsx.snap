// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render text with default styling 1`] = `
<Styled(Text)>
  Default text
</Styled(Text)>
`;

exports[`Text should render text with descriptive styling 1`] = `
<Styled(Text)
  descriptive={true}
>
  Descriptive text
</Styled(Text)>
`;

exports[`Text should render text with strong styling 1`] = `
<Styled(Text)
  strong={true}
>
  Strong text
</Styled(Text)>
`;
